src/EnergyPlus/OutdoorAirUnit.cc

```diff
@@ -1361,8 +1372,7 @@ namespace OutdoorAirUnit {
                                            OutAirUnit(OAUnitNum).OAEquip(compLoop).BranchNum,
                                            OutAirUnit(OAUnitNum).OAEquip(compLoop).CompNum);
                     }
-                    if (static_cast<int>(OutAirUnit(OAUnitNum).OAEquip(compLoop).CoilPlantTypeOfNum) ==
-                        static_cast<int>(CompType::WaterCoil_CoolingHXAsst)) {
+                    if (static_cast<int>(OutAirUnit(OAUnitNum).OAEquip(compLoop).CoilType) == static_cast<int>(CompType::WaterCoil_CoolingHXAsst)) {
```

Comment:
Why do you have to `static_cast<int>` both sides of this comparison? Will it not work if you don't cast either side?

Reply:
This comparison would have been wrong. Thanks for catching this. Since the same coils are listed in both [`CompType`](https://github.com/NREL/EnergyPlus/blob/287ccc7c2f762b39e67bd29bd7ab009d2e41a888/src/EnergyPlus/OutdoorAirUnit.hh#L72-L89) and [`DataPlant::PlantEquipmentType`](https://github.com/NREL/EnergyPlus/blob/c56f48bdd11ea362a06a274d4466be3f5bc0ace7/src/EnergyPlus/Plant/Enums.hh#L95) enum classes but don't have the same int values associated with them. I corrected it. I would have loved to remove the `CoilType` variable altogether but it is used in the [ScanPlantLoopsForObject() function](https://github.com/NREL/EnergyPlus/blob/287ccc7c2f762b39e67bd29bd7ab009d2e41a888/src/EnergyPlus/OutdoorAirUnit.cc#L1231-L1233).

Reply:
I see. Yes, having these partially overlapping enumerations is a problem. This is a fine solution for now. No need to hold up the PR for this specific issue.